impl/sql/mutations/mutations.go

```diff
@@ -77,35 +61,61 @@ func New(db *sql.DB, mapID int64) (mutator.Mutation, error) {
 	return m, nil
 }
 
-// Read reads all mutations for a specific given mapID, epoch, and index.
-func (m *mutations) Read(ctx context.Context, txn transaction.Txn, epoch int64, index []byte) ([]byte, error) {
+// ReadRange reads all mutations for a specific given mapID and sequence range.
+// The range is identified by a starting sequence number and a count.
+func (m *mutations) ReadRange(txn transaction.Txn, startSequence uint64, count int) ([]*tpb.SignedKV, error) {
 	readStmt, err := txn.Prepare(readExpr)
 	if err != nil {
 		return nil, err
 	}
 	defer readStmt.Close()
+	rows, err := readStmt.Query(m.mapID, startSequence, count)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
 
-	var mutation []byte
-	err = readStmt.QueryRow(m.mapID, epoch, index).Scan(&mutation)
-	if err == sql.ErrNoRows {
-		return nil, nil
-	} else if err != nil {
+	var results []*tpb.SignedKV
```

Comment:
This could be slightly improved to avoid dynamic array-resizing at runtime:
`results := make([]*tpb.SignedKV, 0, count)`